lib/phase/merge.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const Hoek = require('@hapi/hoek');
+const TEMPLATE_NAME_REGEX_WITH_NAMESPACE =
+    require('screwdriver-data-schema').config.regex.FULL_TEMPLATE_NAME_WITH_NAMESPACE;
+
+/**
+ * Fetch pipeline template
+ * @method fetchPipelineTemplate
+ * @method
+ * @param   {String}                          pipelineTemplateName                        Pipeline template name
+ * @param   {PipelineTemplateVersionFactory}  config.pipelineTemplateVersionFactory       PiplineTemplateVersion Factory to get pipeline templates
+ * @param   {PipelineTemplateTagFactory}      config.pipelineTemplateTagFactory           PipelineTemplateTag Factory to get pipeline templates tag
+ * @param   {TemplateMetaFactory}             config.templateMetaFactory                  TemplateMeta Factory to get pipeline templates
```

Comment:
we do not use `TemplateMetaFactory`, we need to use `pipelineTemplateFactory`

Reply:
https://github.com/screwdriver-cd/models/blob/41293848c21da3fc419993c2188f969fdde44d39/lib/pipelineTemplateVersionFactory.js#L217

Reply:
`getWithMetadata` uses `TemplateMetaFactory`

Reply:
that is the base class, you need to pass `pipelineTemplateFactory` instead